[docs] Add warning for `luxon` macro tokens (#8245)
mui/mui-x: docs/data/date-pickers/overview/overview.md

- [Luxon](https://moment.github.io/luxon/#/)
- [Moment.js](https://momentjs.com/)

:::warning
The Date and Time Pickers are not working well with Luxon macro-token (`D`, `DD`, `T`, `TT`, ...),
because of [how they are expanded](https://github.com/mui/mui-x/issues/7615).

If your application is only built be a single local, the easiest solution is to manually [provide a format](/x/react-date-pickers/adapters-locale/#custom-formats) that does not contain any macro-token
(e.g. `M/d/yyyy` instead of `D` for the english locale).

> We are trying to find a solution with the Luxon team to improve the built-in support in the near future.

Reviewer: ```suggestion
As soon as a solution is found the built-in support will be improved.
```
